ark/schema/shared/jsonSchema.ts

```diff
@@ -34,51 +37,90 @@ export declare namespace JsonSchema {
 		examples?: readonly t[]
 	}
 
-	export type Branch = Constrainable | Const | String | Numeric | Object | Array
+	type Composition = Union | OneOf | Intersection | Not
+
+	type NonBooleanBranch =
+		| Constrainable
+		| Const
+		| Composition
+		| Enum
+		| String
+		| Numeric
+		| Object
+		| Array
+
+	export type Branch = boolean | JsonSchema
 
 	export interface Constrainable extends Meta {
 		type?: listable<TypeName>
 	}
 
+	export interface Intersection extends Meta {
+		allOf: readonly JsonSchema[]
+	}
+
+	export interface Not {
+		not: JsonSchema
+	}
+
+	export interface OneOf extends Meta {
```

Comment:
Since you already created your own set of types for this, can we remove the changes here?

The existing types are meant to represent the JSON schema ArkType will actually generate, so I'd rather not include stuff like Not.

Reply:
> Since you already created your own set of types for this, can we remove the changes here?

I'm not quite sure what you mean by "created your own set of types for this".

Originally "my" json schema types [the ones for json schema -> arktype] and "your" json schema types [the ones for arktype -> json schema] were completely separate, but I interpreted [your comment above](https://github.com/arktypeio/arktype/pull/1159#discussion_r1788628897) to mean you wanted these types to be shared, and so that's what I've done.

`@ark/jsonschema` now _doesn't_ have its own types -- it just uses the `@ark/schema` JSON Schema types.

Reply:
I wonder if there is a straightforward way then to separate them so that structures we never generate like `OneOf` wouldn't be included in the output type but were included for the input type?

Potentially some kind of recursive filtering, but I'm also happy to take a look at this after it is merged.

Reply:
In the interest of getting an "MVP" through for the work you're looking to do in 2.2, I'll leave this as an exercise for later.